client/components/InputDialog.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+	<div id="input-dialog-overlay" :class="{opened: data !== null}">
+		<div v-if="data !== null" id="input-dialog">
+			<div class="input-text">
+				<div class="input-text-title">{{ data.title }}</div>
+				<p>{{ data.text }}</p>
+				<form
+					id="inputform"
+					class="inputdialogform"
+					method="post"
+					action=""
+					@submit.prevent="onSubmit"
+				>
+					<textarea
+						id="ReasonData"
+						ref="ReasonData"
+						class="input inputDialogBox"
+						:placeholder="data.placeholder"
+						type="text"
+						:value="reason"
+						@input="setReason"
+						@keypress.enter.exact.prevent="close"
+					/>
+				</form>
+			</div>
+			<div class="input-buttons">
+				<button class="btn btn-cancel" @click="close()">Cancel</button>
+				<button class="btn btn-confirm" @click="close(true)">{{ data.button }}</button>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style>
+#input-dialog {
+	background: var(--body-bg-color);
+	color: #fff;
+	margin: 10px;
+	border-radius: 5px;
+	max-width: 500px;
```

Comment:
Do we want this to still be modal-like on mobile? Maybe worth full-screening or something.